src/controllers/userController.js

```diff
@@ -0,0 +1,163 @@
+require('dotenv').config();
+const admin = require('../config/firebase');
+const db = admin.firestore();
+const axios = require('axios');
+
+exports.signupUser = async (req, res) => {
+    // console.log(req.body)
+    const { name, email, password, IsAdmin } = req.body;
```

Comment:
I would highly advocate for renaming "admin" everywhere to "teacher". The two are synonymous in how we think about them, but admin implies "I can do anything". I can send a request to this URL any time (without a token, since they won't have one before signup) and make myself an admin, which sounds terrifying until you remember it is just a "teacher" and has access to your own courses and such.

We also should try to stick to one language or the other to avoid confusion.

Reply:
I will rename "admin" everywhere to "teacher".